### PR TITLE
additional x-forwarded-for handling

### PR DESCRIPTION
**Pull request by blockpane**
- conditionally use alternate header
- default to req.ip
- strip port if present

### CODE REVIEW

File: server/routes/wallet.js [hunk 1]
```diff
@@ -248,10 +248,24 @@ router.post('/public-api/buy-address', handler(async (req, res) => {
...
     ipAddress = ipAddress.split(',')[0]
+  } else if (ipAddress === '') {
+    ipAddress = req.ip
+  }
+  // strip the port if present.
+  const stripPort = ipAddress.split(':')
+  if (stripPort.length === 2) {
+    ipAddress = stripPort[0]
+  } else if (stripPort.length === 7) {
+    ipAddress = stripPort.pop().join(':')
```
Comment (andreyvEze) [inline server/routes/wallet.js:266]: `stripPort.pop().join(':')` would not work. do you want to get rid from last element and then join? in this case you need something like this:
```
stripPort.pop()
ipAddress = stripPort.join(':')
```

File: server/routes/wallet.js [hunk 2]
```diff
@@ -248,10 +248,24 @@ router.post('/public-api/buy-address', handler(async (req, res) => {
   } = req.body
   const processor = await plugins.payment
 
-  let ipAddress = req.headers[process.env.IP_HEADER_PROP_NAME] || ''
+  let ipAddress = ''
+  if (Boolean(process.env.TRUST_PROXY) && process.env.IP_HEADER_PROP_NAME) {
```
Comment (andreyvEze) [inline server/routes/wallet.js:252]: I think you could just check for `process.env.TRUST_PROXY` without `Boolean(process.env.TRUST_PROXY)`

File: server/routes/wallet.js [hunk 3]
```diff
@@ -248,10 +248,24 @@ router.post('/public-api/buy-address', handler(async (req, res) => {
...
+    ipAddress = req.headers[process.env.IP_HEADER_PROP_NAME.toLowerCase()]
+  }
+  // first address of xff, list is comma separated
   if (ipAddress && ipAddress.indexOf(',') > -1) {
     ipAddress = ipAddress.split(',')[0]
+  } else if (ipAddress === '') {
+    ipAddress = req.ip
```
Comment (andreyvEze) [inline server/routes/wallet.js:259]: Are you sure this gets IP from `X-Forwarded-For`? or it should be such value

```
When the trust proxy setting does not evaluate to false, the value of this property is derived from the left-most entry in the X-Forwarded-For header. This header can be set by the client or by the proxy.
```

Comment (blockpane) [reply server/routes/wallet.js:259]: I think this is the behavior we want, if TRUST_PROXY is truthy but IP_HEADER_PROP_NAME value isn't set ipAddress ends up being null, causing errors. Before, if TRUST_PROXY was not set we would alse end up with a null value. Express defaults to x-forwarded-for, and I have confirmed that x-forwarded-for is the default in UAT.

In the end this should prevent having a null IP being set, falling back to what express thinks is the IP. But it does raise a good point that there should be a warning in the logs if it couldn't get the IP.

Comment (andreyvEze) [reply server/routes/wallet.js:259]: ok, just to confirm

**Review by andreyvEze — APPROVED** [review]

**Pull request merged**